src/utils/settings.cpp

```diff
@@ -203,7 +203,12 @@ void multipass::Settings::set_aux(const QString& key, QString val) // work with
 {
     // TODO we should have handler callbacks instead
     if (key == petenv_key && !mp::utils::valid_hostname(val.toStdString()))
-        throw InvalidSettingsException{key, val, "Invalid hostname"};
+    {
+        if (val.isEmpty() && key != "client.primary-name") // Handle "multipass set client.primary-name=". Further checks required.
+        {
+            throw InvalidSettingsException{key, val, "Invalid hostname"};
+        }
+    }
```

Comment:
Hmm, I don't think this is right. You want to accept empty values, so you should just add that condition to the first `if`:

```c++
if (key == petenv_key && !val.isEmpty() && !mp::utils::valid_hostname(val.toStdString()))
    throw ...
```

Note that `petenv_key` is the constant defining `"client.primary-name"` as the key we use for that.

Reply:
Corrected, I had not looked up the `constants` prior.